sdk/src/types/block/output/feature/staking.rs

```diff
@@ -50,23 +49,70 @@ impl StakingFeature {
     }
 }
 
-pub(crate) mod dto {
+mod dto {
     use serde::{Deserialize, Serialize};
 
+    use super::*;
     use crate::utils::serde::string;
 
-    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    #[derive(Serialize, Deserialize)]
     #[serde(rename_all = "camelCase")]
-    pub struct StakingFeatureDto {
+    struct StakingFeatureDto {
         #[serde(rename = "type")]
-        pub kind: u8,
+        kind: u8,
         #[serde(with = "string")]
-        pub staked_amount: u64,
+        staked_amount: u64,
         #[serde(with = "string")]
-        pub fixed_cost: u64,
+        fixed_cost: u64,
         #[serde(with = "string")]
-        pub start_epoch: u64,
+        start_epoch: u64,
         #[serde(with = "string")]
-        pub end_epoch: u64,
+        end_epoch: u64,
+    }
+
+    impl From<&StakingFeature> for StakingFeatureDto {
+        fn from(value: &StakingFeature) -> Self {
+            Self {
+                kind: StakingFeature::KIND,
+                staked_amount: value.staked_amount,
+                fixed_cost: value.fixed_cost,
+                start_epoch: value.start_epoch,
+                end_epoch: value.end_epoch,
+            }
+        }
+    }
+
+    impl From<StakingFeatureDto> for StakingFeature {
+        fn from(value: StakingFeatureDto) -> Self {
+            Self::new(
+                value.staked_amount,
+                value.fixed_cost,
+                value.start_epoch,
+                value.end_epoch,
+            )
+        }
+    }
+
+    impl<'de> Deserialize<'de> for StakingFeature {
+        fn deserialize<D: serde::Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+            let dto = StakingFeatureDto::deserialize(d)?;
+            if dto.kind != Self::KIND {
+                return Err(serde::de::Error::custom(format!(
```

Comment:
Check `no_std` compatability
cannot find macro `format` in this scope

Reply:
dangit